python/ray/serve/_private/replica.py

```diff
@@ -221,6 +207,35 @@ async def initialize_replica():
             # Used to guard `initialize_replica` so that it isn't called twice.
             self._replica_init_lock = asyncio.Lock()
 
+        def _configure_logger_and_profilers(
+            self, replica_tag: ReplicaTag, logging_config: LoggingConfig
+        ):
+            replica_name = ReplicaName.from_replica_tag(replica_tag)
+            if replica_name.app_name:
+                component_name = (
+                    f"{replica_name.app_name}_{replica_name.deployment_name}"
+                )
+            else:
+                component_name = f"{replica_name.deployment_name}"
```

Comment:
Non-blocker: I feel this can be moved into `ReplicaName` as a property. Also feel that delimiter in the `ReplicaName` can be designed better to accommodate this usecase. But those are all out of scope of this PR and feel free to leave as is.

Reply:
Yes planning to do all of that, but this PR will get cherry-picked into release so I want to keep it as minimal as possible